addons/account/models/account_payment.py

```diff
@@ -179,19 +179,23 @@ def _seek_for_lines(self):
         '''
         self.ensure_one()
 
-        liquidity_lines = self.env['account.move.line']
-        counterpart_lines = self.env['account.move.line']
-        writeoff_lines = self.env['account.move.line']
+        # liquidity_lines, counterpart_lines, writeoff_lines
+        lines = [self.env['account.move.line'] for _dummy in range(3)]
 
         for line in self.move_id.line_ids:
             if line.account_id in self._get_valid_liquidity_accounts():
-                liquidity_lines += line
+                lines[0] += line  # liquidity_lines
             elif line.account_id.internal_type in ('receivable', 'payable') or line.account_id == line.company_id.transfer_account_id:
-                counterpart_lines += line
+                lines[1] += line  # counterpart_lines
             else:
-                writeoff_lines += line
+                lines[2] += line  # writeoff_lines
 
-        return liquidity_lines, counterpart_lines, writeoff_lines
+        for i, line in enumerate(lines[:-1]):
+            if not line and len(lines[2]) == 1:
+                lines[i] = lines[2]
+                lines[2] -= lines[2]
```

Comment:
```suggestion
        if len(lines[2]) == 1:
            for i in (0, 1):
                if not lines[i]:
                    lines[i] = lines[2]
                    lines[2] -= lines[2]
                    break
```
Add also a comment to explain the purpose of that :)